### PR TITLE
Json-LF decoding: numeric bounds checking, timestamp truncation and error reporting

### PR DESCRIPTION
**Pull request by raphael-speyer-da**
Also renamed `decimal` decoder to `numeric` and added a bunch of unit tests, especially the error cases.

### CODE REVIEW

**Review by lucianojoublanc-da — APPROVED** [review]
Would suggest getting a review by a second pair of eyes, but looks good to me.

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/json/JsonLfReader.java
Comment (cocreature) [inline language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/json/JsonLfReader.java:37]: what is deprecated here?

**Review by cocreature — APPROVED** [review]
nice thanks!

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/json/JsonLfReader.java
Comment (raphael-speyer-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/json/JsonLfReader.java]: No longer needed, so removed.

**Pull request merged**
